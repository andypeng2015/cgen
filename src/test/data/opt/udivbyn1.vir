module foo

export function w $foo(w %x) {
@start:
  %x = udiv.w %x, 0xffffffff_w
  ret %x
}
