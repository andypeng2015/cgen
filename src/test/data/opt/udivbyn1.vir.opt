module foo

export function w $foo(w %x) {
@2:
  %2 = eq.w %x, 0xffffffff_w ; @6
  %3 = flag.w %2 ; @7
  ret %3
}
