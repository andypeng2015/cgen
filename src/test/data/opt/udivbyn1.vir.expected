module foo

export function w $foo(w %x) {
@0:
  %1 = eq.w %x, 0xffffffff_w ; @3
  %0 = flag.w %1 ; @2
  ret %0
}
